File: java/visitor/RemoteWorkShiftScheduleManagement.java
package visitor;

public class RemoteWorkShiftScheduleManagement implements ScheduleManagement {
    @Override
    public void generateReport() {
        System.out.println("Generating report for remote work shift...");
    }

    @Override
    public void calculateOverTime() {
        System.out.println("Calculating over time for remote work shift..");
    }

    @Override
    public void accept(ScheduleManagementVisitor scheduleManagementVisitor) {
        scheduleManagementVisitor.visit(this);
    }

}

File: C#/Visitor/RemoteWorkShiftScheduleManagement.cs
namespace Visitor;

public sealed class RemoteWorkShiftScheduleManagement : IScheduleManagement
{
    public void CalculateOverTime()
        => Console.WriteLine("Calculating over time for remote work shift..");

    public void GenerateReport()
        => Console.WriteLine("Generating report for remote work shift...");

    public void Accept(IScheduleManagementVisitor scheduleManagementVisitor)
        => scheduleManagementVisitor.Visit(this);
}
